infoschema/slow_log.go

```diff
@@ -127,12 +138,30 @@ func ParseSlowLog(tz *time.Location, scanner *bufio.Scanner) ([][]types.Datum, e
 			}
 		}
 	}
-	if err := scanner.Err(); err != nil {
-		return nil, errors.AddStack(err)
-	}
 	return rows, nil
 }
 
+func getOneLine(reader *bufio.Reader) ([]byte, error) {
+	lineByte, isPrefix, err := reader.ReadLine()
+	var tempLine []byte
+	if err != nil {
+		return lineByte, err
+	}
+	for isPrefix {
+		tempLine, isPrefix, err = reader.ReadLine()
+		lineByte = append(lineByte, tempLine...)
+
+		// Use the max value of max_allowed_packet to check the single line length.
+		if len(lineByte) > int(variable.MaxOfMaxAllowedPacket) {
```

Comment:
Why using MaxOfMaxAllowedPacket instead of MaxAllowedPacket?

Reply:
Because `MaxAllowedPacket` may has been changed.